src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/mappers/TransitLayerUpdater.java

```diff
@@ -99,11 +106,14 @@ public void update(Set<Timetable> updatedTimetables) {
         }
       }
 
-      transitLayer.replaceTripPatternsForDate(
+      realtimeTransitLayer.replaceTripPatternsForDate(
           date,
           new ArrayList<>(patternsForDateMap.values())
       );
 
+      // Switch out the reference with the updated transit layer. This is an atomic operation.
+      graph.realtimeTransitLayer = realtimeTransitLayer;
```

Comment:
Considering what I wrote above, I would change these comments to say "The JMM guarantees reference assignment is atomic, but does not guarantee that the combined process of assigning field contents and publishing the reference is atomic. We should soon replace this assignment with a more strictly correct implementation."

Reply:
As I commented above, I wrapped the realtimeTransitLayer in ConcurrentPublished<T>, so this change should not be needed anymore.